src/components/mdx-components/component-links.tsx

```diff
@@ -1,144 +1,145 @@
-          github.url || `${githubRepoUrl}/tree/main/packages/${github.package}`
-        }
-        icon={FaGithub}
-        iconColor={iconColor}
-        iconSize='1rem'
-      >
-        {t('component.mdx-components.component-links.view-source')}
-      </ComponentLink>
-    </WrapItem>
-  )
-
-  const npmLink = npm?.package && (
-    <WrapItem>
-      <ComponentLink
-        url={`https://www.npmjs.com/package/${npm.package}`}
-        icon={FaNpm}
-        iconSize='2rem'
-        iconColor='red.500'
-      >
-        {npm.package}
-      </ComponentLink>
-    </WrapItem>
-  )
-
-  const storybookLink = storybook?.url && (
-    <WrapItem>
-      <ComponentLink
-        url={storybook.url}
-        icon={StorybookIcon}
-        iconSize='1.25rem'
-        iconColor='pink.500'
-      >
-        {t('component.mdx-components.component-links.view-storybook')}
-      </ComponentLink>
-    </WrapItem>
-  )
-
-  const videoLink = video?.url && (
-    <WrapItem>
-      <ComponentLink
-        url={video.url}
-        icon={FaYoutube}
-        iconSize='1.2rem'
-        iconColor='red.500'
-      >
-        {t('component.mdx-components.component-links.view-video')}
-      </ComponentLink>
-    </WrapItem>
-  )
-
-  const themeComponentLink = theme && (
-    <WrapItem>
-      <ComponentLink
-        url={`${githubRepoUrl}/tree/main/packages/theme/src/components/${theme.componentName}.ts`}
-        icon={FaGithub}
-        iconColor={iconColor}
-        iconSize='1rem'
-      >
-        {t('component.mdx-components.component-links.view-theme-source')}
-      </ComponentLink>
-    </WrapItem>
-  )
-
-  return (
-    <Wrap className='component-links' mt='2rem' spacing='4' {...rest}>
-      {githubLink}
-      {themeComponentLink}
-      {npmLink}
-      {storybookLink}
-      {videoLink}
-    </Wrap>
-  )
-}
-
-export default ComponentLinks
+import {
+  HStack,
+  Icon,
+  Link,
+  useColorModeValue,
+  ButtonGroup,
+  Button,
+  ButtonProps,
+  Text,
+} from '@chakra-ui/react'
+import React from 'react'
+import { FaGithub, FaNpm, FaYoutube } from 'react-icons/fa'
+import StorybookIcon from '../storybook-icon'
+import { t } from 'utils/i18n'
+
+type ComponentLinkProps = ButtonProps & {
+  icon: React.ElementType
+  url: string
+  iconSize?: string
+  iconColor?: string
+}
+
+function ComponentLink(props: ComponentLinkProps) {
+  const { icon, url, children, iconSize, iconColor, ...rest } = props
+  return (
+    <Button
+      as={Link}
+      href={url}
+      isExternal
+      px='12px'
+      display='flex'
+      alignItems='center'
+      minH='32px'
+      borderWidth='1px'
+      borderRadius='md'
+      color={useColorModeValue('gray.600', 'whiteAlpha.700')}
+      _hover={{
+        color: useColorModeValue('gray.700', 'whiteAlpha.900'),
+        bg: useColorModeValue('gray.300', 'whiteAlpha.300'),
+        boxShadow: 'sm',
+        transform: 'translateY(-1px)',
+        textDecor: 'none',
+      }}
+      {...rest}
+    >
+      <HStack>

```

Comment:
You can use the leftIcon prop of the Button component to place the icons.

Reply:
With the Button component setup, have to use the `sx` prop to ensure the `iconSize` and `iconColor` passed in are used. No major issue there.